src/ui/public/courier/data_source/_migrate_filter.js

```diff
@@ -0,0 +1,19 @@
+import _ from 'lodash';
+
+export function migrateFilter(filter) {
+  if (filter.match) {
+    const fieldName = Object.keys(filter.match)[0];
+    const params = filter.match[fieldName];
+
+    if (_.isPlainObject(params) && params.type === 'phrase') {
```

Comment:
Could it be worth extracting this into a separate predicate?

```javascript
function isMatchPhraseFilter(filter, fieldName) {
  return _.get(filter, ['match', fieldName, 'type']) === 'phrase';
}
```

Then the condition would be very descriptive as to its purpose:

```javascript
  if (isMatchPhraseFilter(filter, fieldName)) {
    // ...
  }
```